src/icu.rs

```diff
@@ -54,11 +54,15 @@ pub fn set_common_data_71(data: &'static [u8]) -> Result<(), i32> {
 
 /// Returns BCP47 language tag.
 pub fn get_language_tag() -> String {
-  let mut output = [0u8; 1024];
-  let len = unsafe {
-    icu_get_default_locale(output.as_mut_ptr() as *mut c_char, output.len())
-  };
-  std::str::from_utf8(&output[..len]).unwrap().to_owned()
+  let mut output = Vec::<u8>::with_capacity(1024);
+  unsafe {
+    let len = icu_get_default_locale(
+      output.spare_capacity_mut().as_mut_ptr().cast::<c_char>(),
```

Comment:
Redudant `spare_capacity_mut`? Just do `output.as_mut_ptr() as *mut c_char`

Reply:
Well, that would have provenance to access the initialised section of the vector, which is empty..

Reply:
```rust
    #[stable(feature = "vec_spare_capacity", since = "1.60.0")]
    #[inline]
    pub fn spare_capacity_mut(&mut self) -> &mut [MaybeUninit<T>] {
        // Note:
        // This method is not implemented in terms of `split_at_spare_mut`,
        // to prevent invalidation of pointers to the buffer.
        unsafe {
            slice::from_raw_parts_mut(
                self.as_mut_ptr().add(self.len) as *mut MaybeUninit<T>,
                self.buf.capacity() - self.len,
            )
        }
    }
```

`self.len` is 0. `self.buf.capacity()` is 1024. So this just becomes:
```rust
slice::from_raw_parts_mut(self.as_mut_ptr(), self.buf.capacity()) // <-- redundant
  .as_mut_ptr()
```

---

Also the example for https://doc.rust-lang.org/beta/std/vec/struct.Vec.html#method.as_mut_ptr
```rust
let mut x: Vec<i32> = Vec::with_capacity(size);
let x_ptr = x.as_mut_ptr();
// ...
```

Reply:
> ```rust
>     #[stable(feature = "vec_spare_capacity", since = "1.60.0")]
>     #[inline]
>     pub fn spare_capacity_mut(&mut self) -> &mut [MaybeUninit<T>] {
>         // Note:
>         // This method is not implemented in terms of `split_at_spare_mut`,
>         // to prevent invalidation of pointers to the buffer.
>         unsafe {
>             slice::from_raw_parts_mut(
>                 self.as_mut_ptr().add(self.len) as *mut MaybeUninit<T>,
>                 self.buf.capacity() - self.len,
>             )
>         }
>     }
> ```
> 
> `self.len` is 0. `self.buf.capacity()` is 1024. So this just becomes:
> ```rust
> slice::from_raw_parts_mut(self.as_mut_ptr(), self.buf.capacity()) // <-- redundant
>   .as_mut_ptr()
> ```

This is implementation detail, and I don't yet know whether the sliced layout is stable & usable for consumers of the stdlib...

> 
> ---
> 
> Also the example for https://doc.rust-lang.org/beta/std/vec/struct.Vec.html#method.as_mut_ptr
> ```rust
> let mut x: Vec<i32> = Vec::with_capacity(size);
> let x_ptr = x.as_mut_ptr();
> // ...
> ```

Now, this is great! Additionally, it shows that the layout is allowed to be relied upon, it seems.